__tests__/cron/exportToTinybird.ts

```diff
@@ -0,0 +1,345 @@
+import { DataSource } from 'typeorm';
+import { saveFixtures } from '../helpers';
+import createOrGetConnection from '../../src/db';
+import { ArticlePost, PostTag, Source, User } from '../../src/entity';
+import { sourcesFixture } from '../fixture/source';
+import { postsFixture, postTagsFixture } from '../fixture/post';
+import {
+  PostsMetadataRepositoryDependency,
+  PostsRepositoryDependency,
+  PostsMetadataRepository,
+  PostsRepository,
+  TinybirdExportService,
+  TinybirdPost,
+} from '../../src/cron/exportToTinybird';
+import {
+  ITinybirdClient,
+  TinybirdDatasourceMode,
+  PostDatasourceResult,
+  QueryResult,
+  fetchfn,
+  TinybirdClient,
+} from '../../src/common/tinybird';
+import { FastifyBaseLogger } from 'fastify';
+
+let con: DataSource;
+
+beforeAll(async () => {
+  con = await createOrGetConnection();
+});
+
+beforeEach(async () => {
+  jest.clearAllMocks();
+
+  await saveFixtures(con, Source, sourcesFixture);
+  await saveFixtures(con, ArticlePost, postsFixture);
+  await saveFixtures(con, PostTag, postTagsFixture);
+  await con
+    .getRepository(User)
+    .save({ id: '1', name: 'Ido', image: 'https://daily.dev/ido.jpg' });
+  await con.getRepository(User).save({
+    id: '2',
+    name: 'Lee',
+    image: 'https://daily.dev/lee.jpg',
+  });
+});
+
+describe('PostsRepository', () => {
+  it('should return posts to export to tinybird with specific properties', async () => {
+    const now = new Date();
+    const latest = new Date(now.getTime() - 10000);
+    const repo = new PostsRepository(con);
+
+    const posts = await repo.getForTinybirdExport(latest);
+    posts.forEach((post) => {
+      post.created_at = '';
+      post.metadata_changed_at = '';
+    });
+    expect(posts).toMatchSnapshot();
+  });
+});
+
+describe('PostsMetadataRepository', () => {
+  const dataSource = 'datasource_mock';
+
+  it('append success', async () => {
+    const tinybirdMock = {
+      postToDatasource: async (
+        datasource: string,
+        mode: TinybirdDatasourceMode,
+        csv: string,
+      ): Promise<PostDatasourceResult> => {
+        expect(datasource).toEqual(dataSource);
+        expect(mode).toEqual(TinybirdDatasourceMode.APPEND);
+        expect(csv).toMatchSnapshot();
+
+        return {} as PostDatasourceResult;
+      },
+    } as ITinybirdClient;
+
+    const postsMetadataRepository = new PostsMetadataRepository(
+      tinybirdMock,
+      dataSource,
+    );
+
+    const posts: TinybirdPost[] = [
```

Comment:
I think we can also use ` await saveFixtures(con, ArticlePost, postsFixture);` something like this right?

Reply:
This exact test is not intended to use a real database